core/log/file_console_logger.odin

```diff
@@ -31,44 +34,158 @@ Default_File_Logger_Opts :: Options{
+	- To make a logger that writes to an `^os2.File` refer to `make_file_logger_os2`
+	- To make a logger that writes to `stdout` and `stderr` refer to `make_console_logger`
+*/
 File_Console_Logger_Data :: struct {
-	file_handle:  os.Handle,
+	type: File_Type,
+	file_handle_os: os.Handle,
+	file_handle_os2: ^os2.File,
 	ident: string,
+	close_file_on_delete: bool,
+	allocator: runtime.Allocator,
 }
 
-create_file_logger :: proc(h: os.Handle, lowest := Level.Debug, opt := Default_File_Logger_Opts, ident := "") -> Logger {
-	data := new(File_Console_Logger_Data)
-	data.file_handle = h
+/*
+Makes a new logger that will write to the provided `os.Handle`
+
+*Allocates Using Provided Allocator*
+
+Inputs:
+- h: An os handle that the logger will write to
+- lowest: The lowest level logging to accept
+- opt: The wanted logging options
+- ident: An identifier that will be written alongside the logged message
+- close_file_on_delete: Sets the logger to close the handle when delete_file_logger is called if `true`
+- allocator: (default: context.allocator)
+- loc: The caller location for debugging purposes (default: `#caller_location`)
+
+Returns:
+- res: The new file logger 
+- res: An allocator error if one occured, `nil` otherwise 
+*/
+make_file_logger_os :: proc(h: os.Handle, lowest := Level.Debug, opt := Default_File_Logger_Opts, ident := "", close_file_on_delete := false, allocator := context.allocator, loc := #caller_location) -> (res: Logger, err: runtime.Allocator_Error) {
+	data := new(File_Console_Logger_Data, allocator, loc) or_return
+	data.type = .os
+	data.file_handle_os = h
 	data.ident = ident
-	return Logger{file_console_logger_proc, data, lowest, opt}
+	data.allocator = allocator
+	data.close_file_on_delete = close_file_on_delete
+	return Logger{file_console_logger_proc, data, lowest, opt}, nil
 }
 
-destroy_file_logger :: proc(log: Logger) {
-	data := cast(^File_Console_Logger_Data)log.data
-	if data.file_handle != os.INVALID_HANDLE {
-		os.close(data.file_handle)
-	}
-	free(data)
+/*
+Makes a new logger that will write to the provided `^os2.File`
+
+*Allocates Using Provided Allocator*
+
+Inputs:
+- f: An os2 file that the logger will write to
+- lowest: The lowest level logging to accept
+- opt: The wanted logging options
+- ident: An identifier that will be written alongside the logged message
+- close_file_on_delete: Sets the logger to close the file when delete_file_logger is called if `true`
+- allocator: (default: context.allocator)
+- loc: The caller location for debugging purposes (default: `#caller_location`)
+
+Returns:
+- res: The new file logger 
+- res: An allocator error if one occured, `nil` otherwise 
+*/
+make_file_logger_os2 :: proc(f: ^os2.File, lowest := Level.Debug, opt := Default_File_Logger_Opts, ident := "", close_file_on_delete := false, allocator := context.allocator, loc := #caller_location) -> (res: Logger, err: runtime.Allocator_Error) {
```

Comment:
What's the reason for implementing the of closing the file on the logger's side? Creating a logger isn't what created the file so it's weird that the responsibility for closing it falls onto the logger. Removing this feature will simplify the implementation by a lot, and closing it on the user's side is as simple as simply calling the appropriate function.

Reply:
It's to retain the old logger behavior until it's removed. I'll probably make it privately set by the create* functions instead.

Reply:
Leaking a file descriptor on logger destroy shouldn't have terrible side effects. I think it'd make sense to simplify the code right now and not worry about the breakage.